src/libslic3r/GCode.cpp

```diff
@@ -3942,65 +3942,134 @@ std::string GCode::_before_extrude(const ExtrusionPath &path, const std::string
+                if (m_config.support_material_acceleration.value >= 0)
+                    acceleration = std::min(max_acceleration, m_config.support_material_acceleration.get_abs_value(acceleration));
+                break;
+            case erSupportMaterialInterface:
+                if (m_config.support_material_interface_acceleration.value >= 0)
+                    acceleration = std::min(max_acceleration, m_config.support_material_interface_acceleration.get_abs_value(acceleration));
+                break;
+            case erBridgeInfill:
+                if (m_config.bridge_acceleration.value >= 0)
+                    acceleration = std::min(max_acceleration, m_config.bridge_acceleration.get_abs_value(acceleration));
+            case erInternalBridgeInfill:
+                if (m_config.bridge_acceleration_internal.value >= 0)
+                    acceleration = std::min(max_acceleration, m_config.bridge_acceleration_internal.get_abs_value(acceleration));
+                break;
+            case erOverhangPerimeter:
+                if (m_config.overhangs_acceleration.value >= 0)
+                    acceleration = std::min(max_acceleration, m_config.overhangs_acceleration.get_abs_value(acceleration));
+                break;
+            case erGapFill:
+                if (m_config.gap_fill_acceleration.value >= 0)
+                    acceleration = std::min(max_acceleration, m_config.gap_fill_acceleration.get_abs_value(acceleration));
+            case erThinWall:
+                if (m_config.thin_walls_acceleration.value >= 0)
+                    acceleration = std::min(max_acceleration, m_config.thin_walls_acceleration.get_abs_value(acceleration));
+            case erSkirt:
+            case erWipeTower:
+            case erMilling:
+            case erCustom:
+            case erMixed:
+            case erCount:
+            case erNone:
+            default:
+                break;
+        }
+
         if (this->on_first_layer() && m_config.first_layer_acceleration.value > 0) {
-            acceleration = std::min(max_acceleration, m_config.first_layer_acceleration.get_abs_value(acceleration));
-        } else if (m_config.perimeter_acceleration.value > 0 && is_perimeter(path.role())) {
-            acceleration = std::min(max_acceleration, m_config.perimeter_acceleration.get_abs_value(acceleration));
-        } else if (m_config.bridge_acceleration.value > 0 && is_bridge(path.role())
-            && path.role() != erOverhangPerimeter) {
-            acceleration = std::min(max_acceleration, m_config.bridge_acceleration.get_abs_value(acceleration));
-        } else if (m_config.infill_acceleration.value > 0 && is_infill(path.role())) {
-            acceleration = std::min(max_acceleration, m_config.infill_acceleration.get_abs_value(acceleration));
+            acceleration = std::min(max_acceleration, std::min(acceleration, m_config.first_layer_acceleration.get_abs_value(acceleration)));
```

Comment:
if travel, it's maybe better to make the min with `m_config.first_layer_acceleration.get_abs_value(travel_acceleration)`

Reply:
Travel acceleration is calculated further down. I think at the moment it is not limited to first layer acceleration. I will add that.